src/components/LanguageSwitch/LanguageSwitch.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { MouseEvent } from 'react';
+import LanguageIcon from '@material-ui/icons/Language';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import i18next, { TFunction } from 'i18next';
+import { useTranslation } from 'react-i18next';
+import { withStyles } from '@material-ui/core/styles';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import Grow from '@material-ui/core/Grow';
+import Popper from '@material-ui/core/Popper';
+import MenuList from '@material-ui/core/MenuList';
+import styled from '@emotion/styled';
+
+import Paper from '../../muiComponents/Paper';
+import MenuItem from '../../muiComponents/MenuItem';
+import Button from '../../muiComponents/Button';
+import Tooltip from '../../muiComponents/Tooltip';
+import Divider from '../../muiComponents/Divider';
+import Box from '../../muiComponents/Box';
+import { Theme } from '../../design-tokens/theme';
+import Link from '../Link';
+import Icon from '../Icon';
+
+const getTranslatedCurrentLanguage = (
+  t: TFunction
+): { [key: string]: { translation: string; icon: React.ComponentProps<typeof Icon>['name'] } } => ({
+  'en-us': {
+    translation: t('lng.english'),
+    icon: 'usa',
+  },
+  'pt-br': {
+    translation: t('lng.portuguese'),
+    icon: 'brazil',
+  },
+  'de-de': {
+    translation: t('lng.german'),
+    icon: 'germany',
+  },
+  'es-es': {
+    translation: t('lng.spanish'),
+    icon: 'spain',
+  },
+  'zh-cn': {
+    translation: t('lng.chinese'),
+    icon: 'china',
+  },
+});
+
+const LanguageSwitch = () => {
+  const { t } = useTranslation();
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef<HTMLButtonElement>(null);
+
+  const languages = i18next.options.resources ? Object.keys(i18next.options.resources) : [];
+  const currentLanguage = i18next.language || i18next.options?.fallbackLng?.[0];
+
+  const { translation: userLanguage } = getTranslatedCurrentLanguage(t)[currentLanguage.toLowerCase()];
+
+  const handleToggle = () => {
```

Comment:
we might optimize the callbacks with `useCallback`, what do you think?

Reply:
I think it's valid :)